scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+        wsrep_log_error "Killing SST ($CLONE_INSTANCE_PID) with SIGKILL after stalling for ${WSREP_SST_IDLE_TIMEOUT} seconds."
+        wsrep_log_error "Within the last ${WSREP_SST_IDLE_TIMEOUT} seconds (defined by the sst-idle-timeout variable),"
+        wsrep_log_error "the SST process on the joiner (this node) has not received any data from the donor."
+        wsrep_log_error "This error could be caused by broken network connectivity between"
+        wsrep_log_error "the donor and the joiner (this node)."
+        wsrep_log_error "Check the network connection and restart the joiner node."
+        wsrep_log_error "Line $LINENO"
+        wsrep_log_error "****************************************************** "
+        exit $ETIMEDOUT
+    fi
+
+    wsrep_log_info "clone recepient daemon finished"
+
+    CLONE_INSTANCE_PID=""
+
+    # Execute first restart for
+    wsrep_log_info "FIRST RESTART to fix the dictionary"
+    wsrep_log_info "Performing data recovery"
+    wsrep_log_debug "-> RECOVERY COMMAND LINE: $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS --wsrep_provider=none"
+
+
+    # Remove created clone user and SST pxc user
+    wsrep_log_debug "-> CLEAN OR NOT? $CLEANUP_CLONE_PLUGIN"
+    CLEANUP_CLONE_PLUGIN_SQL=""
+    if [ "${CLEANUP_CLONE_PLUGIN}" == "yes" ]; then
+            CLEANUP_CLONE_PLUGIN_SQL="UNINSTALL PLUGIN CLONE;"
+    fi
+cat << EOF > "$CLONE_SQL"
+SET SESSION sql_log_bin=OFF;
+DROP USER IF EXISTS $CLONE_USER;
+DROP USER IF EXISTS $CLONE_USER@'localhost';
+DROP USER IF EXISTS 'mysql.pxc.sst.user'@'localhost';
+$CLEANUP_CLONE_PLUGIN_SQL
+SHUTDOWN;
+EOF
+    
+    eval $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS --wsrep_provider=none --init_file="$CLONE_SQL"
+
+    wsrep_log_info "SHUTDOWN done"
+    wsrep_log_info "Second restart for recovery position"
+    wsrep_log_debug "-> Second restart: $CLONE_BINARY $DEFAULT_OPTIONS $SKIP_NETWORKING_OPTIONS --wsrep_recover"
+    eval $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS $SKIP_NETWORKING_OPTIONS --wsrep_recover >> $CLONE_ERR 2>&1
+
+    RP="$(grep -a '\[WSREP\] Recovered position:' $CLONE_ERR || :)"
+    RP_PURGED=`echo $RP | sed 's/.*WSREP\]\ Recovered\ position://' | sed 's/^[ \t]*//'`
+    wsrep_log_info "Recovered POSITION: [$RP_PURGED]"
+
+    # If an invalid recovery position is returned we report the one we get from Donor
+    if [ "$RP_PURGED" == "00000000-0000-0000-0000-000000000000:-1" ]; then
+        RP_PURGED=$RP_PURGED_EMERGENCY
+    fi
+    if [ ! "$RP_PURGED" == "" ]; then
+
+        ORIG_IFS=$IFS
+        IFS=':'
+        read -ra gtid_arr <<< "$RP_PURGED"
+        uuid="${gtid_arr[0]}"
+        position="${gtid_arr[1]}"
+        IFS=$ORIG_IFS
+        wsrep_log_debug "-> Creating grastate.dat file in $GRASTATE_FILE UUID: $uuid POSITION: $position"
+cat << EOF > "$GRASTATE_FILE"
+# GALERA saved state
+version: 2.1
+uuid:    $uuid
+seqno:   $position
+safe_to_bootstrap: 0
+EOF
+
+    fi
+    if [ "$?" == "0" ]; then
```

Comment:
To which command does the `$?` refer to here?

Reply:
Reworked this part. It contained a dead code.